### PR TITLE
[CA] Register recently evicted pods in NodeDeletionTracker.

### PR DESCRIPTION
**Pull request by olagacek**
#### Which component this PR applies to?

cluster-autoscaler
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Part of parallel scale down implementation.

### CODE REVIEW

File: cluster-autoscaler/core/scaledown/actuation/drain.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 	policyv1beta1 "k8s.io/api/policy/v1beta1"
 	kube_errors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/core/scaledown/deletiontracker"
```
Comment (x13n) [inline cluster-autoscaler/core/scaledown/actuation/drain.go:30]: I would consider dropping dependency on deletion tracker and just accepting a simple local interface in `NewDefaultEvictor`.

**Review by x13n — COMMENTED** [review]
Thanks for the change! Just two minor comments. I don't think they are blocking, so LGTMing the change and putting on hold for you to consider.

/lgtm
/hold

File: cluster-autoscaler/core/scaledown/actuation/drain.go (hunk 1 above)
Comment (olagacek) [reply cluster-autoscaler/core/scaledown/actuation/drain.go:30]: to double check I understood it correctly: you would like me to introduce a new local interface with RegisterEviction function (implemented by node deletion tracker)?

Comment (olagacek) [reply cluster-autoscaler/core/scaledown/actuation/drain.go]: created new interface

File: cluster-autoscaler/core/scaledown/actuation/drain_test.go [hunk 2]
```diff
@@ -441,8 +442,8 @@ func TestDrainNodeWithPodsEvictionFailure(t *testing.T) {
 	}
 	ctx, err := NewScaleTestAutoscalingContext(options, fakeClient, nil, nil, nil, nil)
 	assert.NoError(t, err)
-
-	evictor := Evictor{EvictionRetryTime: 0, PodEvictionHeadroom: DefaultPodEvictionHeadroom}
+	ndt := deletiontracker.NewNodeDeletionTracker(DefaultPodEvictionHeadroom)
```
[the remark this answers is not in the document]
Comment (olagacek) [reply cluster-autoscaler/core/scaledown/actuation/drain_test.go]: removed

**Pull request merged**